src/Host/API/Impl/Definitions/IRHostSession.cs

```diff
@@ -0,0 +1,179 @@
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task StopHostAsync(bool waitForShutdown = true, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Attempts to cancel all running tasks in the R Host. 
+        /// This is similar to 'Interrupt R' command.
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task CancelAllAsync(CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Executes R code
+        /// </summary>
+        /// <param name="expression">Expression or block of R code to execute</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <exception cref="ArgumentNullException" />
+        /// <exception cref="REvaluationException" />
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task ExecuteAsync(string expression, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Evaluates the provided expression and returns the result.
+        /// This method is typically used to fetch variable value and return it to .NET code.
+        /// </summary>
+        /// <typeparam name="T">
+        /// Type of the variable expected. This must be a simple type.
+        /// To return collections use <see cref="GetListAsync"/> and <see cref="GetDataFrameAsync"/>
+        /// </typeparam>
+        /// <param name="expression">Expression or block of R code to execute</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>The variable or expression value</returns>
+        /// <exception cref="ArgumentException" />
+        /// <exception cref="REvaluationException" />
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task<T> EvaluateAsync<T>(string expression, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Invokes R function. 
+        /// </summary>
+        /// <param name="function">Function name</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <param name="args">Function arguments</param>
+        /// <exception cref="ArgumentException" />
+        /// <exception cref="REvaluationException" />
+        /// <exception cref="OperationCanceledException" />
+        /// <exception cref="RHostDisconnectedException" />
+        Task InvokeAsync(string function, CancellationToken cancellationToken = default(CancellationToken), params object[] args);
```

Comment:
`function` is a name (i.e no arguments). Yes, one can do `EvalAsync("c(1:10)") but how to pass arguments from C#? `InvokeAsync` constructs R call from function name and arguments passed as .NET types. I.e. InvokeAsync("x", new List<int>() {1,2,3})` turns into `EvalAsync("x(c(1, 2, 3))")`